### DISCUSSION TITLE
Moving instance labels to another layer

### DISCUSSION THREAD

**Discussion by jan-david-fischbach** [gdsfactory/gdsfactory#404] [Q&A] [answered]
Assuming I want to build my own PDK with different layers than for example the UBCPDK. How would I change the layer that `add_instance_label` writes its label to? I can not just setup the right layers in the PDK TECH, because `add_instance_label` uses TECH from gf.tech, right?

**Comment by joamatab** [comment]
the current way of doing this is using functools.partial [see tutorial](https://gdsfactory.github.io/gdsfactory/notebooks/_0_python.html#partial)

```
add_instal_label_jan = gf.partial(gf.add_labels, layer_label=(3,0))

```

Related to #405

**Reply by jan-david-fischbach** [reply]
I agree partial is a great option to make this work. I am in a place, where using it seems a bit difficult to me: I use circuit components defined from yaml in my PDK. These are registered by `PDK.register_cells_yaml` which uses `from_yaml`, which then uses `add_instance_label`. I am not sure where to specify the layer in this configuration

**Reply by tvt173** [reply]
it would be great if we created a layer of abstraction for layers in the PDK, similar to what we have done recently with cross sections. then we could refer to the label layer as `"INSTANCE_LABEL"`, and that could be overriden in your PDK to whichever layer number/datatype you desire. that would automatically fix the default `add_instance_label` function to work for your PDK.

for more complex scenarios, it would also be nice if you could register a default decorator. then you could override how the cells are decorated entirely, if needed

**Reply by jan-david-fischbach** [reply]
agree

**Comment by joamatab** [comment, marked as answer]
just added this functionality of using strings registered in the PDK in https://github.com/gdsfactory/gdsfactory/pull/413/

and made a minor release gdsfactory 5.8.0

let me know what you think
Joaquin
